cassandra/src/main/java/io/micronaut/configuration/cassandra/CassandraSessionFactory.java

```diff
@@ -0,0 +1,312 @@
+        configurations.put(option.getPath(), Duration.ofMillis(value.orElse(defaultValue)));
+    }
+
+    /**
+     * Inserts configuration from application context into {@link com.typesafe.config.Config} mapping path from cassandra.datasource.* to {@link DefaultDriverOption} path by type.
+     * Maps int to duration by milliseconds.
+     *
+     * @param configurations map of paths from {@link com.typesafe.config.Config} to values to use in cassandra driver
+     * @param option         driver option from cassandra driver
+     */
+    private void putDurationMilliseconds(Map<String, Object> configurations, String prefix, DefaultDriverOption option) {
+        Optional<Integer> value = this.resolver.getProperty(CassandraConfiguration.PREFIX + "." + prefix + "." + option.getPath(), int.class);
+        value.ifPresent(t -> configurations.put(option.getPath(), Duration.ofMillis(t)));
+    }
+
+    /**
+     * Inserts configuration from application context into {@link com.typesafe.config.Config} mapping path from cassandra.datasource.* to {@link DefaultDriverOption} path by type.
+     * Maps int to duration by nanoseconds with default value if not present in {@link PropertyResolver}.
+     *
+     * @param configurations map of paths from {@link com.typesafe.config.Config} to values to use in cassandra driver
+     * @param option         driver option from cassandra driver
+     */
+    private void putDurationNanoSeconds(Map<String, Object> configurations, String prefix, DefaultDriverOption option, int defaultValue) {
+
+        Optional<Integer> value = this.resolver.getProperty(CassandraConfiguration.PREFIX + "." + prefix + "." + option.getPath(), int.class);
+        configurations.put(option.getPath(), Duration.ofNanos(value.orElse(defaultValue)));
+    }
+
+    /**
+     * Inserts configuration from application context into {@link com.typesafe.config.Config} mapping path from cassandra.datasource.* to {@link DefaultDriverOption} path by type.
+     * Maps int to duration by nanoseconds.
+     */
+    private void putDurationNanoSeconds(Map<String, Object> configurations, String prefix, DefaultDriverOption option) {
+        Optional<Integer> value = this.resolver.getProperty(CassandraConfiguration.PREFIX + "." + prefix + "." + option.getPath(), int.class);
+        value.ifPresent(t -> configurations.put(option.getPath(), Duration.ofNanos(t)));
+    }
+
+    /**
+     * Creates the {@link CqlSessionBuilder} bean for the given configuration.
+     *
+     * @param configuration The cassandra configuration bean
+     * @return A {@link CqlSession} bean
+     */
+    @EachBean(CassandraConfiguration.class)
+    public CqlSessionBuilder session(CassandraConfiguration configuration) {
+
+        try {
+            CqlSessionBuilder builder = CqlSession.builder().withConfigLoader(new DefaultDriverConfigLoader(() -> {
```

Comment:
This is not really maintainable. We would need to find a way to handle this automatically

Reply:
umm, I've tried a couple different things such as creating those properties in CassandraConfiguration but they tended to be a lot more complex and cumbersome. The other option is do something like this.

```
ConfigFactory.invalidateCaches()
val ref = ConfigFactory.load().getConfig("datastax-java-driver")
val app: Config = ref.getConfig('other-prefix');
app.withFallback(ref)
```

application.conf
```
other-prefix {
    basic.config-reload-interval = 5 minutes
}
datastax-java-driver {
    basic.config-reload-interval = 5 minutes
}
```

Reply:
I found out I could just pass the map of properties directly.